OsmAnd/src/net/osmand/plus/settings/fragments/HistoryItemsFragment.java

```diff
@@ -264,7 +270,12 @@ public void onCategorySelected(List<Object> items, boolean selected) {
 
 	@Override
 	public void onDeletionConfirmed() {
-		deleteSelectedItems();
+		DeleteHistoryTask deleteHistoryTask = new DeleteHistoryTask(requireActivity(), getDeleteHistoryType(), selectedItems, this);
```

Comment:
We added async deletion; won't it lead to some errors or crashes in SearchHistoryHelper?